File: uiweb/src/visual/MiniCAstGraphEdge.ts
import type { JavaMirrorFile } from "../translation/javaMirror";

export const miniCAstGraphEdgeMirror = {
  javaPath: "src/main/java/minic/uilocal/visual/MiniCAstGraphEdge.java",
  webPath: "uiweb/src/visual/MiniCAstGraphEdge.ts",
  packageName: "minic.uilocal.visual",
  exportName: "MiniCAstGraphEdge",
  kind: "record",
  imports: ["java.util.Objects"],
  fields: [],
  methods: [{ name: "MiniCAstGraphEdge", signature: "MiniCAstGraphEdge(String fromId, String toId, double fromX, double fromY, double toX, double toY, boolean hot)" }],
} as const satisfies JavaMirrorFile;

export class MiniCAstGraphEdge {
  static readonly mirror = miniCAstGraphEdgeMirror;

  constructor(
    readonly fromId: string,
    readonly toId: string,
    readonly fromX: number,
    readonly fromY: number,
    readonly toX: number,
    readonly toY: number,
    readonly hot: boolean,
  ) {
    if (fromId.length === 0 || toId.length === 0) {
      throw new Error("edge endpoints must not be empty");
    }
  }
}

export default MiniCAstGraphEdge;

File: src/main/java/minic/uilocal/visual/MiniCAstGraphEdge.java
package minic.uilocal;

import java.util.Objects;

/**
 * AST 图中的连线。
 *
 * @param fromId 父节点 ID
 * @param toId 子节点 ID
 * @param fromX 起点横坐标
 * @param fromY 起点纵坐标
 * @param toX 终点横坐标
 * @param toY 终点纵坐标
 * @param hot 是否连接当前路径
 */
public record MiniCAstGraphEdge(
        String fromId,
        String toId,
        double fromX,
        double fromY,
        double toX,
        double toY,
        boolean hot
) {
    public MiniCAstGraphEdge {
        Objects.requireNonNull(fromId, "fromId");
        Objects.requireNonNull(toId, "toId");
    }
}
